src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -2012,10 +2059,19 @@
   <Node Name="BoundIndexOrRangePatternIndexerAccess" Base="BoundExpression">
     <Field Name="Type" Type="TypeSymbol" Override="true" Null="disallow" />
 
-    <Field Name="Receiver" Type="BoundExpression" Null="disallow" />
-    <Field Name="LengthOrCountProperty" Type="PropertySymbol" Null="disallow" />
-    <Field Name="PatternSymbol" Type="Symbol" Null="disallow" />
+    <!-- An expression with type Index or Range -->
     <Field Name="Argument" Type="BoundExpression" Null="disallow" />
+
+    <Field Name="LengthOrCountAccess" Type="BoundExpression?"/>
```

Comment:
>LengthOrCountAccess [](http://example.com/codeflow?start=17&length=19)

I think we should apply  ```SkipInVisitor="true"``` to this node. For example, SemanticModel shouldn't be visiting it. This comment is applicable to similar node added elsewhere.